v2/apiserver/internal/core/kubernetes/substrate.go

```diff
@@ -643,40 +643,45 @@ func (s *substrate) DeleteWorkerAndJobs(
 		},
 	).AsSelector().String()
 
-	// Delete all pods related to this Event
-	if err := s.kubeClient.CoreV1().Pods(
-		project.Kubernetes.Namespace,
-	).DeleteCollection(
-		ctx,
-		metav1.DeleteOptions{},
-		metav1.ListOptions{
-			LabelSelector: labelSelector,
-		},
-	); err != nil {
-		return errors.Wrapf(
-			err,
-			"error deleting event %q pods in namespace %q",
-			event.ID,
+	// If a worker's phase is CANCELED, it could have only reached this terminal
+	// phase if it was previously in PENDING and hence, no pods would have been
+	// created. Therefore, we just skip to cleaning up the event secret(s) below.
+	if event.Worker.Status.Phase != core.WorkerPhaseCanceled {
```

Comment:
👍

Reply:
Also true for an Event whose Worker is _still_ in a PENDING phase. That could happen if `DeleteWorkerAndJobs` was invoked in response to a `Delete` or `DeleteMany` on the `EventService`.